Cache/Hashing.lean

```diff
@@ -11,10 +11,26 @@ namespace Cache.Hashing
 
 open System IO
 
+/--
+The `HashMemo` contains all information `Cache` needs about the modules:
+* the name
+* its imports
+* the file's hash (in `hashMap` and `cache`)
+
+additionally, it contains the `rootHash` which reflects changes to Mathlib's
+Lake project settings.
+-/
 structure HashMemo where
+  /-- Hash of mathlib's lake project settings. -/
   rootHash : UInt64
+  /-- Maps the `.lean` file of a module to the `.lean` files of its imports. -/
   depsMap  : Std.HashMap FilePath (Array FilePath) := ∅
+  /--
+  For modules in Mathlib or upstream, this contains the same information
+  as `hashMap`. Contains `none` if a source file couldn't be found.
```

Comment:
Is it `none` only for source files not in mathlib?

Reply:
updated the docstring. Files downstream will fall into the latter category, but not necessarily exclusively.

E.g. if you'll write `import Mathlib.Data` in another mathlib file and try to get cache, this file will have `none` as `Mathlib/Data.lean` doesn't exist. (meaning no cache will be retrieved for the file with bad import)